File: test15_out/test15/Msp430g2231/Msp430g2231.c

/*
 * ======== string literals ========
 */

static const char test15_Msp430g2231__s__fileName[] = "Msp430g2231.p";
static const char test15_Msp430g2231__s__0[] = "Msp430g2231";

/*
 * ======== code injections ========
 */

 #include <msp430g2231.h> 

/*
 * ======== function definitions ========
 */

void test15_Msp430g2231_wait__E( uint32 us ) {
}

void test15_Msp430g2231_pollen__run__E() {
}

void test15_Msp430g2231_reset__E() {
    WDTCTL = WDTPW | WDTHOLD;

    BCSCTL1 = CALBC1_1MHZ;

    DCOCTL = CALDCO_1MHZ;

    if ((test15_Msp430g2231_mclkFrequency__V) == 1000000) {
        if ((test15_Msp430g2231_smclkFrequency__V) == 500000) {
            BCSCTL2 = DIVS_1;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 250000) {
            BCSCTL2 = DIVS_2;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 125000) {
            BCSCTL2 = DIVS_3;

        }
    }
    else if ((test15_Msp430g2231_mclkFrequency__V) == 500000) {
        BCSCTL2 = DIVM_1;

        if ((test15_Msp430g2231_smclkFrequency__V) == 250000) {
            BCSCTL2 = DIVS_1;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 125000) {
            BCSCTL2 = DIVS_2;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 62500) {
            BCSCTL2 = DIVS_3;

        }
    }
    else if ((test15_Msp430g2231_mclkFrequency__V) == 250000) {
        BCSCTL2 = DIVM_2;

        if ((test15_Msp430g2231_smclkFrequency__V) == 125000) {
            BCSCTL2 = DIVS_1;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 62500) {
            BCSCTL2 = DIVS_2;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 31250) {
            BCSCTL2 = DIVS_3;

        }
    }
    else if ((test15_Msp430g2231_mclkFrequency__V) == 125000) {
        BCSCTL2 = DIVM_3;

        if ((test15_Msp430g2231_smclkFrequency__V) == 62500) {
            BCSCTL2 = DIVS_1;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 31250) {
            BCSCTL2 = DIVS_2;

        }
        else if ((test15_Msp430g2231_smclkFrequency__V) == 15625) {
            BCSCTL2 = DIVS_3;

        }
    }
    else {
    }
}

void test15_Msp430g2231_targetInit__I() {
}

void test15_Msp430g2231_shutdown__E() {
}

void test15_Msp430g2231_nop__E() {
     asm("nop") ;

}

File: test15_out/test15/Msp430g2231/Msp430g2231.js

/*
 * ======== MODULE test15.Msp430g2231 ========
 */

var $$c = function() {
    const Msp430g2231 = {};
    Msp430g2231.Msp430g2231 = Msp430g2231
    Msp430g2231.$name = 'test15.Msp430g2231';
    Msp430g2231.pollen$used = 0;
    Msp430g2231.reset = new $$Ref('test15_Msp430g2231_reset__E');
    Msp430g2231.shutdown = new $$Ref('test15_Msp430g2231_shutdown__E');
    Msp430g2231.pollen__run = new $$Ref('test15_Msp430g2231_pollen__run__E');
    Msp430g2231.wait = new $$Ref('test15_Msp430g2231_wait__E');
    Msp430g2231.nop = new $$Ref('test15_Msp430g2231_nop__E');
    Msp430g2231.targetInit = new $$Ref('test15_Msp430g2231_targetInit__I');
    Msp430g2231.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Msp430g2231.pollen__uses__ = function() {
    }
    Msp430g2231.$$privateInit = function() {
        Msp430g2231.mclkFrequency = 1000000;
        Msp430g2231.smclkFrequency = 1000000;
        Msp430g2231.aclkFrequency = 0;
    }
    return Msp430g2231;
}
var $$u = $$c();
$units['test15.Msp430g2231'] = $$u;
$units.push($$u);

